migrate branch feat/CP-124-migrate-users-profile-to-turbo

CliffDawen opened the pull request:
Feat: cp 124 migrate user profile to turbo

added update user profile page:
- implemented user profile page by Dawen
- retrieve user profile by Dawen
- implemented update user profile(displayname) by Dawen
- add loading animation and toast for setting page by May
- enhance logic of persist with Zustand by May
- modify the style of the prompt text by Ethan

Solve cp-124 migrate user profile to turbo

1. navigate to user profile page
![image](https://github.com/user-attachments/assets/1f884f51-9447-404a-88e9-878455be8e6c)
2. navigate to update user profile
![image](https://github.com/user-attachments/assets/3a944112-8a25-4f9c-814e-86ff63c0e0a4)
3. update user profile
![image](https://github.com/user-attachments/assets/4a08c811-1ef9-40d3-98f2-0af184897f6b)
4. input validation
![image](https://github.com/user-attachments/assets/42371d6b-d055-4c20-b0e6-43e8e0d9cde0)5. data changed
![image](https://github.com/user-attachments/assets/851aa717-8827-4c32-942f-d2cc4b337e6f)

Chris-Wang commented:
A good job!

But after checking our client, I found two pages are doing the same job:
http://localhost:3010/en/dashboard/profile
http://localhost:3010/en/account/setting
both are doing update on user profile

So I want to add some extra scope in this PR:
1. Revert /profile, we still want a page to show user profile (avatar, user information)
2. The updated url for updating profile should be: http://localhost:3010/en/account/profile/edit, and that page should combine http://localhost:3010/en/dashboard/profile with http://localhost:3010/en/account/setting
or you can just add logic to http://localhost:3010/en/account/setting to make it work 
3. The new added page could be accessed via top menu -> Profile Edit
4. Update the original `Setting` on top menu, now it should be `App Setting`, url: http://localhost:3010/en/appsetting, the same as the one on left bar

Maaayhan committed:
feat: migrate user profile to turbo

Added update user profile page:
-implemented user profile page by Dawen
-retrieve user profile by Dawen
-implemented update user profile(displayname) by Dawen
- add loading animation and toast for setting page by May
- enhance logic of persist with Zustand by May
- modify the style of the prompt text by Ethan

Solve CP-124 migrate user profile to turbo

Chris-Wang reviewed (approved):
LGTM
